tests/unit/lms/views/basic_lti_launch_test.py

```diff
@@ -234,17 +235,14 @@ def test_it_enables_frontend_grading(self, context, pyramid_request):
 
         context.js_config.maybe_enable_grading.assert_called_once_with()
 
-    def test_it_adds_the_document_url(self, context, pyramid_request):
-        mic = factories.ModuleItemConfiguration(
-            resource_link_id=pyramid_request.params["resource_link_id"],
-            tool_consumer_instance_guid=pyramid_request.params[
-                "tool_consumer_instance_guid"
-            ],
-        )
-
+    def test_it_adds_the_document_url(
+        self, assignment_service, context, pyramid_request
+    ):
         db_configured_basic_lti_launch_caller(context, pyramid_request)
 
-        context.js_config.add_document_url.assert_called_once_with(mic.document_url)
+        context.js_config.add_document_url.assert_called_once_with(
+            assignment_service.get_document_url.return_value
+        )
```

Comment:
It was nice when we improved these tests by replacing the previous mock with a factory. Now we've made them even nicer by replacing the factory with a new mock. The difference is that the code that we're mocking and the code that we're testing is now simpler because the service offers a simpler interface than the model classmethods did: it's just `get_document_url()` with no args. That's really easy to write isolated mock-based tests for